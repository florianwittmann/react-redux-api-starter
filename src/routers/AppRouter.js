import React from "react";

import Home from "../scenes/Home";

import Login from "../scenes/User/Login";
import Signup from "../scenes/User/Signup";
import Logout from "../scenes/User/Logout";
import Reset from "../scenes/User/Reset";
import ResetSuccess from "../scenes/User/Reset/success";
import ResetToken from "../scenes/User/Reset/_token";
import Dashboard from "../scenes/Dashboard";

import Header from "../components/Header";
import Footer from "../components/Footer";
import RequireAuthentication from "./RequireAuthentication";
import { BrowserRouter, Route, Switch } from "react-router-dom";

const AppRouter = () => (
  <BrowserRouter>
    <div>
      <Header />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/login" component={Login} />
        <Route path="/signup" component={Signup} />
        <Route path="/logout" component={Logout} />
        <Route path="/reset/success" component={ResetSuccess} />
        <Route path="/reset/:token" component={ResetToken} />
        <Route path="/reset" component={Reset} />
        <Route path="/dashboard" component={RequireAuthentication(Dashboard)} />
      </Switch>
      <Footer />
    </div>

  </BrowserRouter>
);

export default AppRouter;
