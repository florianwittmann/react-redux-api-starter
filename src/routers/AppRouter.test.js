import React from "react";
import { shallow } from "enzyme";
import AppRouter from "./AppRouter";
test("should render the AppRouter", () => {
  const wrapper = shallow(<AppRouter />);
  expect(wrapper).toMatchSnapshot();
});
