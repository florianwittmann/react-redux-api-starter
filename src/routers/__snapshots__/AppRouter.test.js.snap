// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the AppRouter 1`] = `
<BrowserRouter>
  <div>
    <withRouter(Connect(Component)) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/signup"
      />
      <Route
        component={[Function]}
        path="/logout"
      />
      <Route
        component={[Function]}
        path="/reset/success"
      />
      <Route
        component={[Function]}
        path="/reset/:token"
      />
      <Route
        component={[Function]}
        path="/reset"
      />
      <Route
        component={[Function]}
        path="/dashboard"
      />
    </Switch>
    <Component />
  </div>
</BrowserRouter>
`;
